### PR TITLE
Use MicroMutableOpResolver in hello_world test.

### PR DESCRIPTION
**Pull request by tanmaydas82**
BUG=http://b/275073322,http://b/274460695

### CODE REVIEW

File: tensorflow/lite/micro/examples/hello_world/evaluate_test.cc [hunk 1]
```diff
@@ -24,6 +24,20 @@ limitations under the License.
...
+using HelloWorldOpResolver = tflite::MicroMutableOpResolver<1>;
+alignas(alignof(HelloWorldOpResolver)) uint8_t
+    op_resolver_buffer[sizeof(HelloWorldOpResolver)];
+}  // namespace
+
+namespace tflite {
+TfLiteStatus GetOpResolver(HelloWorldOpResolver** op_resolver) {
+  *op_resolver = new (op_resolver_buffer) HelloWorldOpResolver();
+  TF_LITE_ENSURE_STATUS((*op_resolver)->AddFullyConnected());
+  return kTfLiteOk;
```
Comment (advaitjain) [inline tensorflow/lite/micro/examples/hello_world/evaluate_test.cc:37]: there's not a very good reason to do a placement new here.

I would suggest something like:

```cc
// Registers the specific OPs needed for the current model.
TfLiteStatus RegisterOps(HelloWorldOpResolver& op_resolver) {
  TF_LITE_ENSURE_STATUS(op_resolver.AddFullyConnected());
  return kTfLiteOk;
}
```

File: tensorflow/lite/micro/examples/hello_world/evaluate_test.cc [hunk 2]
```diff
@@ -24,6 +24,20 @@ limitations under the License.
 #include "tensorflow/lite/micro/system_setup.h"
 #include "tensorflow/lite/schema/schema_generated.h"
 
+namespace {
+using HelloWorldOpResolver = tflite::MicroMutableOpResolver<1>;
+alignas(alignof(HelloWorldOpResolver)) uint8_t
+    op_resolver_buffer[sizeof(HelloWorldOpResolver)];
+}  // namespace
+
+namespace tflite {
```
Comment (advaitjain) [inline tensorflow/lite/micro/examples/hello_world/evaluate_test.cc:33]: Delete. The OpResolver function can go in the anon namespace.

File: tensorflow/lite/micro/examples/hello_world/evaluate_test.cc [hunk 3]
```diff
@@ -36,14 +50,17 @@ TfLiteStatus LoadFloatModelAndPerformInference() {
         model->version(), TFLITE_SCHEMA_VERSION);
   }
 
-  // This pulls in all the operation implementations we need
-  tflite::AllOpsResolver resolver;
+  HelloWorldOpResolver* op_resolver = nullptr;
+  if (tflite::GetOpResolver(&op_resolver) != kTfLiteOk) {
+    MicroPrintf("Could not resolve all the ops.");
+    return kTfLiteError;
+  }
```
Comment (advaitjain) [inline tensorflow/lite/micro/examples/hello_world/evaluate_test.cc:57]: This should become:
```cc
HelloWorldOpResolver op_resolver;
TF_LITE_ENSURE_STATUS(RegisterOps(op_resolver));
```

We don't need additional error messages.

File: tensorflow/lite/micro/examples/hello_world/evaluate_test.cc [hunk 4]
```diff
@@ -97,14 +114,17 @@ TfLiteStatus LoadQuantModelAndPerformInference() {
...
```
Comment (advaitjain) [inline tensorflow/lite/micro/examples/hello_world/evaluate_test.cc:121]: same comment as above

File: tensorflow/lite/micro/examples/hello_world/evaluate_test.cc [hunk 5]
```diff
@@ -36,14 +50,17 @@ TfLiteStatus LoadFloatModelAndPerformInference() {
...
 
   constexpr int kTensorArenaSize = 2056;
   uint8_t tensor_arena[kTensorArenaSize];
```
Comment (advaitjain) [inline tensorflow/lite/micro/examples/hello_world/evaluate_test.cc:60]: add a comment that the arena size is simply a round number. The exact arena usage can be determined using the RecordingMicroInterpreter. And we should add that to the current .cc file in this or a follow-up PR.

And let's change the number to be 2000 or a suitably large number.

File: tensorflow/lite/micro/examples/hello_world/evaluate_test.cc [hunk 6]
```diff
@@ -97,14 +114,17 @@ TfLiteStatus LoadQuantModelAndPerformInference() {
...
```
Comment (advaitjain) [inline tensorflow/lite/micro/examples/hello_world/evaluate_test.cc:124]: same comment as above

**Review by advaitjain — CHANGES_REQUESTED** [review]

File: tensorflow/lite/micro/examples/hello_world/evaluate_test.cc (hunk 1 above)
Comment (tanmaydas82) [reply tensorflow/lite/micro/examples/hello_world/evaluate_test.cc:37]: done

File: tensorflow/lite/micro/examples/hello_world/evaluate_test.cc (hunk 2 above)
Comment (tanmaydas82) [reply tensorflow/lite/micro/examples/hello_world/evaluate_test.cc:33]: done

File: tensorflow/lite/micro/examples/hello_world/evaluate_test.cc (hunk 3 above)
Comment (tanmaydas82) [reply tensorflow/lite/micro/examples/hello_world/evaluate_test.cc:57]: done

File: tensorflow/lite/micro/examples/hello_world/evaluate_test.cc (hunk 4 above)
Comment (tanmaydas82) [reply tensorflow/lite/micro/examples/hello_world/evaluate_test.cc:121]: done

File: tensorflow/lite/micro/examples/hello_world/evaluate_test.cc (hunk 6 above)
Comment (tanmaydas82) [reply tensorflow/lite/micro/examples/hello_world/evaluate_test.cc:117]: done

**Review by advaitjain — APPROVED** [review]

**Review by advaitjain — APPROVED** [review]

**Pull request merged**